src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -0,0 +1,378 @@
+                        )
+                    );
+                }
+            } catch (CoreException $e) {
+                $this->flashErrors([
+                    $this->trans(
+                        sprintf(
+                            'Could not generate mail templates for theme %s with locale %s',
+                            $data['mailTheme'],
+                            $data['language']
+                        ),
+                        'Admin.Notifications.Error'
+                    ),
+                    $e->getMessage(),
+                ]);
+            }
+        }
+
+        return $this->redirectToRoute('admin_mail_theme_index');
+    }
+
+    /**
+     * Save mail theme configuration
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Exception
+     */
+    public function saveConfigurationAction(Request $request)
+    {
+        /** @var FormHandlerInterface $formHandler */
+        $formHandler = $this->getMailThemeFormHandler();
+        /** @var Form $form */
+        $form = $formHandler->getForm()->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            if (!$form->isValid()) {
+                $this->flashErrors($this->getFormErrorsForJS($form));
+
+                return $this->redirectToRoute('admin_mail_theme_index');
+            }
+
+            $errors = $formHandler->save($form->getData());
+            if (empty($errors)) {
+                $this->addFlash(
+                    'success',
+                    $this->trans(
+                        'Mail theme configuration saved successfully',
+                        'Admin.Notifications.Success'
+                    )
+                );
+            } else {
+                $this->flashErrors($errors);
+            }
+        }
+
+        return $this->redirectToRoute('admin_mail_theme_index');
+    }
+
+    /**
+     * Preview the list of layouts for a defined theme
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     * @param string $themeName
+     *
+     * @return Response
+     *
+     * @throws InvalidArgumentException
+     */
+    public function previewThemeAction(Request $request, $themeName)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+
+        /** @var ThemeCatalogInterface $themeCatalog */
+        $themeCatalog = $this->get('prestashop.core.mail_template.theme_catalog');
+        /** @var ThemeInterface $mailTheme */
+        $mailTheme = $themeCatalog->getByName($themeName);
+
+        return $this->render('@PrestaShop/Admin/Improve/Design/MailTheme/preview.html.twig', [
+            'layoutHeaderToolbarBtn' => [],
+            'layoutTitle' => $this->trans('Preview Theme %s', 'Admin.Navigation.Menu', [$mailTheme->getName()]),
```

Comment:
Same question, menu entry or tab?

Reply:
This one is not in the menu, it's a "subpage" This is just the title page when you want details about a theme

Reply:
I would localize it in _Admin.Design.Feature_ then.

Reply:
Ok, makes sense!